src/site/content/en/reliable/offline-cookbook/index.md

```diff
@@ -0,0 +1,761 @@
+description: Some common recipes for making your app work offline.
+authors:
+  - jakearchibald
+date: 2014-12-09
+updated: 2020-09-28
+---
+
+With [Service Worker][sw_primer] we gave up trying to solve offline, and gave developers the moving
+parts to go solve it themselves. It gives you control over caching and how requests are handled.
+That means you get to create your own patterns. Let's take a look at a few possible patterns in
+isolation, but in practice you'll likely use many of them in tandem depending on URL and context.
+
+For a working demo of some of these patterns, see [Trained-to-thrill][ttt], and
+[this video](https://www.youtube.com/watch?v=px-J9Ghvcx4) showing the performance impact.
+
+## The cache machine—when to store resources
+
+[Service Worker][sw_primer] lets you handle requests independently from caching, so we'll look at
+them separately. First up, caching, when should it be done?
+
+### On install—as a dependency {: #on-install-as-dependency }
+
+<figure class="w-figure">
+  <img class="w-screenshot" alt="" src="cm-on-install-dep.png">
+</figure>
+
+Service Worker gives you an `install` event. You can use this to get stuff ready, stuff that must be
+ready before you handle other events. While this happens any previous version of your Service Worker
+is still running and serving pages, so the things you do here mustn't disrupt that.
+
+**Ideal for:** CSS, images, fonts, JS, templates… basically anything you'd consider static to that
+"version" of your site.
+
+These are things that would make your site entirely non-functional if they failed to fetch, things
+an equivalent native-app would make part of the initial download.
+
+```js
+self.addEventListener('install', function (event) {
+  event.waitUntil(
+    caches.open('mysite-static-v3').then(function (cache) {
+      return cache.addAll([
+        '/css/whatever-v3.css',
+        '/css/imgs/sprites-v6.png',
+        '/css/fonts/whatever-v8.woff',
+        '/js/all-min-v4.js',
+        // etc.
+      ]);
+    }),
+  );
+});
+```
+
+`event.waitUntil` takes a promise to define the length and success of the install. If the promise
+rejects, the installation is considered a failure and this Service Worker will be abandoned (if an
+older version is running, it'll be left intact). `caches.open` and `cache.addAll` return promises.
+If any of the resources fail to fetch, the `cache.addAll` call rejects.
+
+On [trained-to-thrill][ttt] I use this to
+[cache static assets](https://github.com/jakearchibald/trained-to-thrill/blob/3291dd40923346e3cc9c83ae527004d502e0464f/www/static/js-unmin/sw/index.js#L3).
+
+### On install—not as a dependency {: #on-install-not }
+
+<figure class="w-figure">
+  <img class="w-screenshot" alt="" src="cm-on-install-not.png">
+</figure>
+
+Similar to above, but won't delay install completing and won't cause installation to fail if caching
+fails.
+
+**Ideal for:** Bigger resources that aren't needed straight away, such as assets for later levels of
```

Comment:
```suggestion
**Ideal for:** bigger resources that aren't needed straight away, such as assets for later levels of
```

Reply:
The explanation is rather esoteric. I don't expect anyone to absorb the reasons for this. In case you're curious, here it is.

The standard in English it to use lower case after a colon. (Notes are an exception.) There's a second rule that applies in this case, the run-in heading ("Ideal for") and the subsequent clause function together in a grammatically valid way. In other words if I added a subject you would have a valid English sentence: "This is ideal for...". This is a valid exception to the rule (there aren't many) to always use complete sentences.

Reply:
Wow, thanks for the explanation!